app/master/subjob.py

```diff
@@ -85,12 +87,28 @@ def api_representation(self):
         }
 
     @property
-    def atoms(self):
+    def atoms(self) -> List['Atom']:
         """
-        :rtype: list[app.master.atom.Atom]
+        Returns a list of all atoms for this subjob
         """
         return self._atoms
 
+    def get_atoms(self, offset: int=None, limit: int=None) -> List['Atom']:
+        """
+        Returns a list of atoms for this subjob
+        :param offset: The starting index of the requested build
+        :param limit: The number of builds requested
+        :rtype: list[app.master.atom.Atom]
+        """
+        num_atoms = len(self._atoms)
+        start, end = get_paginated_indices(offset, limit, num_atoms)
+
+        # Offset request/starting index is out of bounds, so return no results.
+        if start > num_atoms:
```

Comment:
For `islice` the out of bounds index will throw an exception, but for a list, partitioning it like that shouldn't cause any problems so we can remove that check here, nice catch